[PT-789] publish sailthru react scripts

kryptx opened the pull request:
### Context
For the content blocks project, we have set up a [new react application](https://github.com/sailthru/content-blocks-frontend) with Facebook's [create-react-app](https://github.com/facebook/create-react-app). Prior to now, Sailthru teams have quickly [ejected](https://github.com/sailthru/content-feed-frontend/commit/9ea5878550fe54bc6fd04111ea0e59a1ee757527) their CRA applications in order to customize webpack configurations. The CRA team [recommends forking react-scripts instead](https://create-react-app.dev/docs/alternatives-to-ejecting/) to perform these customizations. This prevents teams from having to manage their webpack configurations themselves; we can keep using react-scripts as a dev-friendly layer to hide the complexity of webpack and pals, and still get our customized build packages. 🎂

EXG suggested exploring this idea now to determine whether we would benefit from it. Since then I have confirmed, using a local `file:` dependency, that changes in this repository can be made to impact the build output of `react-scripts build`. 

Note that we have forked **create-react-app**, which is a [lerna](https://lerna.js.org/) project. We (like most forks of this repository) are only interested in modifying and publishing a customized version of the `react-scripts` package.

As a setup step to using our own fork of react-scripts, we will need to be able to publish it to a package repository so that our applications can depend on it. Since the `main` branch is considered unstable, we expect a tag. The tag should point to a stable version that has not been published within sailthru.

### This PR
Adds a workflow which runs on demand and expects a tag, e.g. `v4.0.3`. The workflow will:

* fetch upstream (facebook) changes and tags
* check out the tagged revision
* merge the `sailthru` branch into it ([example changes against v4.0.3](https://github.com/sailthru/create-react-app/compare/v4.0.3...sailthru))
* release the module (scoped to sailthru) in GitHub packages

If the merge on this step fails, it means this repository's `main` branch needs to be updated from upstream. The `sailthru` branch should be rebased on top of it.

If at any time we urgently need to make changes to our customization, but the latest version is the same as the version we have published, we should create and publish a pre-release for the next patch version allowing work to continue.

kryptx committed:
[PT-789] Add release workflow for @sailthru/react-scripts

kryptx committed:
[PT-789] Publish requested version with merge

kryptx committed:
[PT-789] Don't require repo to be up to date

The longer it has been since pulling upstream, the slower builds will be.
Eventually, the workflow will have a merge conflict. At that point,
we should update main from upstream and rebase our changes against it.

dylannz-sailthru commented:
~Can we add a little more to the PR description? Why do we (and other teams) want this change?~ thanks :)

kryptx commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,82 @@
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        NPM_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        NODE_AUTH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+      run: yarn release
```
Just realized this needs to change. I copied this from stui-scripts. It's not a yarn command, it's a script implemented in that repo. Will update.

cgreene-st commented:
@kryptx wondering if the longer term view might be to import the stui scripts here since this is now a "sailthru" version of CRA?

kryptx commented:
> wondering if the longer term view might be to import the stui scripts here since this is now a "sailthru" version of CRA?

Yes, I think so. It feels to me that stui-scripts should be a _peer_ dependency in this fork -- but we will need to ensure that its own dependency versions are compatible with the versions that this fork will install. I've made no effort to do that yet.

kryptx committed:
[PT-789] Remove install/build/test scripts

- The test and build scripts do not exist in react-scripts
- We assume stable versions
- Install doesn't verify much of anything on its own
- Install takes time

m1dcarry reviewed (commented):
LGTM
